tofu/tests/tests02_data/test_04_core_new.py

```diff
@@ -0,0 +1,703 @@
+
+"""
+This module contains tests for tofu.geom in its structured version
+"""
+
+# Built-in
+import os
+import warnings
+
+# Standard
+import numpy as np
+import scipy.constants as scpct
+import matplotlib.pyplot as plt
+
+# tofu-specific
+from tofu import __version__
+import tofu.data as tfd
+import tofu.utils as tfu
+
+_here = os.path.abspath(os.path.dirname(__file__))
+VerbHead = 'tofu.data.DataCollection'
+
+
+#######################################################
+#
+#     Setup and Teardown
+#
+#######################################################
+
+
+def setup_module(module):
+    print("")   # this is to get a newline after the dots
+    LF = os.listdir(_here)
+    lss = ['TFD_', 'Test', '.npz']
+    LF = [lf for lf in LF if all([ss in lf for ss in lss])]
+    LF = [
+        lf for lf in LF
+        if not lf[lf.index('_Vv')+2:lf.index('_U')] == __version__
+    ]
+    print("Removing the following previous test files:")
+    print(LF)
+    for lf in LF:
+        os.remove(os.path.join(_here, lf))
+    # print("setup_module before anything in this file")
+
+
+def teardown_module(module):
+    # os.remove(VesTor.Id.SavePath + VesTor.Id.SaveName + '.npz')
+    # os.remove(VesLin.Id.SavePath + VesLin.Id.SaveName + '.npz')
+    # print("teardown_module after everything in this file")
+    # print("") # this is to get a newline
+    LF = os.listdir(_here)
+    lss = ['TFD_', 'Test', '.npz']
+    LF = [lf for lf in LF if all([ss in lf for ss in lss])]
+    LF = [
+        lf for lf in LF
+        if lf[lf.index('_Vv')+2:lf.index('_U')] == __version__
+    ]
+    print("Removing the following test files:")
+    print(LF)
+    for lf in LF:
+        os.remove(os.path.join(_here, lf))
+    pass
+
+
+# def my_setup_function():
+#    print ("my_setup_function")
+
+# def my_teardown_function():
+#    print ("my_teardown_function")
+
+# @with_setup(my_setup_function, my_teardown_function)
+# def test_numbers_3_4():
+#    print 'test_numbers_3_4  <============================ actual test code'
+#    assert multiply(3,4) == 12
+
+# @with_setup(my_setup_function, my_teardown_function)
+# def test_strings_a_3():
+#    print 'test_strings_a_3  <============================ actual test code'
+#    assert multiply('a',3) == 'aaa'
+
+
+#######################################################
+#
+#     Creating Ves objects and testing methods
+#
+#######################################################
+
+
+class Test01_DataCollection(object):
```

Comment:
Just out of curiosity: is there a reason why you chose to use classes for the tests ?

Reply:
Its partly because it legacy code from a tutorial for nosetest, and partly because I find it handy since it allows to pass data from test to tests using the class attributes